pkg/kubelet/util/pluginwatcher/example_plugin.go

```diff
@@ -0,0 +1,157 @@
+	resp := new(v1beta1.ExampleResponse)
+	return resp, nil
+}
+
+func (s *pluginServiceV1Beta1) RegisterService() {
+	v1beta1.RegisterExampleServer(s.server.grpcServer, s)
+}
+
+type pluginServiceV1Beta2 struct {
+	server *examplePlugin
+}
+
+func (s *pluginServiceV1Beta2) GetExampleInfo(ctx context.Context, rqt *v1beta2.ExampleRequest) (*v1beta2.ExampleResponse, error) {
+	glog.Infof("GetExampleInfo v1beta2_field: %s", rqt.V1Beta2Field)
+	resp := new(v1beta2.ExampleResponse)
+	return resp, nil
+}
+
+func (s *pluginServiceV1Beta2) RegisterService() {
+	v1beta2.RegisterExampleServer(s.server.grpcServer, s)
+}
+
+// NewExamplePlugin returns an initialized examplePlugin instance
+func NewExamplePlugin() *examplePlugin {
+	return &examplePlugin{
+		stop: make(chan bool),
+	}
+}
+
+// NewTestExamplePlugin returns an initialized examplePlugin instance for testing
+func NewTestExamplePlugin() *examplePlugin {
+	return &examplePlugin{
+		stop:               make(chan bool),
+		registrationStatus: make(chan watcherapi.RegistrationStatus),
+	}
+}
+
+// GetInfo is the RPC invoked by plugin watcher
+func (e *examplePlugin) GetInfo(stream watcherapi.Registration_GetInfoServer) error {
+	info := watcherapi.PluginInfo{
+		Type:    "example-plugin-type",
+		Name:    "example-plugin",
+		Version: []string{"v1beta1", "v1beta2"},
+	}
+	err := stream.Send(&info)
+	if err != nil {
+		glog.Errorf("Error sending plugin info: %v\n", err)
+		return err
+	}
+
+	s, err := stream.Recv()
+	if err != nil {
+		glog.Errorf("Received err: %v\n", err)
+		return err
+	}
+	glog.Infof("Got stream")
+	fmt.Println("stream:", s)
+	if e.registrationStatus != nil {
+		glog.Infof("Got stream1")
+		e.registrationStatus <- *s
+		glog.Infof("Got strea2")
+		glog.Infof("%v\n", e.registrationStatus)
+	}
+	if s.PluginRegistered {
+		glog.Infof("Registration success!!\n")
+		return nil
+	} else {
+		glog.Errorf("Registration failed: %s\n", s.Error)
+		return fmt.Errorf("Registration failed: %s", s.Error)
+	}
+}
+
+// Serve starts example plugin grpc server
+func (e *examplePlugin) Serve(socketPath string) error {
+	glog.Infof("starting example server at: %s\n", socketPath)
+	lis, err := net.Listen("unix", socketPath)
+	if err != nil {
+		glog.Errorf("starting example server failed: %v", err)
+		return err
+	}
+	glog.Infof("example server started at: %s\n", socketPath)
+	e.grpcServer = grpc.NewServer()
+	// Registers kubelet plugin watcher api.
+	watcherapi.RegisterRegistrationServer(e.grpcServer, e)
+	// Registers services for both v1beta1 and v1beta2 versions.
+	v1beta1 := &pluginServiceV1Beta1{server: e}
+	v1beta1.RegisterService()
+	v1beta2 := &pluginServiceV1Beta2{server: e}
+	v1beta2.RegisterService()
+
+	// Starts service
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		// Blocking call to accept incoming connections.
+		err := e.grpcServer.Serve(lis)
+		glog.Errorf("example server stopped serving: %v", err)
+	}()
+
+	<-e.stop
```

Comment:
What happens if the server fails to start? 
Would be better to start the server, and then wait for either:
- server error
- server stop chan